codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -359,20 +479,27 @@ Returns Error if the node with given id cannot be found (unexpected)
 
 newItems are ordered earliest to latest (as they are returned from the worker)
 */
-let update = (self: t, ~id, ~latestFetchedBlock, ~newItems, ~currentBlockHeight): result<
-  t,
-  exn,
-> => {
-  let isFetchingAtHead =
-    currentBlockHeight <= latestFetchedBlock.blockNumber ? true : self.isFetchingAtHead
-  self
-  ->updateInternal(
-    ~id,
-    ~latestFetchedBlock,
-    ~reversedNewItems=newItems->Array.reverse,
-    ~isFetchingAtHead,
-  )
-  ->Result.map(result => pruneAndMergeNextRegistered(result)->Option.getWithDefault(result))
+let update = (
+  {baseRegister, pendingDynamicContracts, isFetchingAtHead}: t,
+  ~id,
+  ~latestFetchedBlock,
+  ~newItems,
+  ~currentBlockHeight,
+): result<t, exn> => {
+  let isFetchingAtHead = isFetchingAtHead || currentBlockHeight <= latestFetchedBlock.blockNumber
+
+  baseRegister
+  ->updateInternal(~id, ~latestFetchedBlock, ~reversedNewItems=newItems->Array.reverse)
+  ->Result.map(updatedRegister => {
+    let withNewDynamicContracts =
+      updatedRegister->addDynamicContractRegisters(pendingDynamicContracts)
+    let maybeMerged = withNewDynamicContracts->pruneAndMergeNextRegistered
+    {
+      baseRegister: maybeMerged->Option.getWithDefault(withNewDynamicContracts),
+      pendingDynamicContracts: [],
+      isFetchingAtHead,
+    }
```

Comment:
Update now first: 
- adds the new items to the targeted register,
- then registers any pending dynamic contracts
- then merges registers

This ensures registers remain ordered based on latestFetched block and can't be messed around by in flight queries